widget/IntersectionBetweenModelErrors.tsx

```diff
@@ -791,8 +594,43 @@ class IntersectionBetweenModelErrors extends Component<IntersectionBetweenModelE
   }
 
   render() {
+    const diagramInfo = "Displays the newly trained model’s error counts and percentages in relation to those of the previous model.";
+    const progressInfo = "Indicates errors made by the previous model that the newly trained model does not make.​";
+    const regressInfo = "Indicates errors made by the newly trained model that the previous model did not make.​";
+    const intersectionInfo = "Indicates errors made by both the previous and newly trained models.​";
+    const _this = this;
+    function getBlockClass(regionName: string) : string {
+      if (_this.state.regionSelected == regionName) {
+        return "venn-legend-row-block-selected";
+      } else {
+        return "venn-legend-row-block";
+      }
+    }
+
     return (
       <div className="plot plot-venn" ref={this.node} id="venndiagramplot">
+        <div className="plot-title-row">
```

Comment:
Would it be possible to abstract away the Legend and its tooltips into its own component? VennLegend or something? You can leave it in the same file and put it near the top.

Then this part of the code could be simplified to just have the `<VennLegend />` and its associated props and methods set on it.